aten/src/ATen/ScalarType.h

```diff
@@ -121,10 +121,10 @@ static inline ScalarType promoteTypes(ScalarType a, ScalarType b) {
     /* u1 */ { u1, i2, i2, i4, i8, f2, f4, f8, ud },
     /* i1 */ { i2, i1, i2, i4, i8, f2, f4, f8, ud },
     /* i2 */ { i2, i2, i2, i4, i8, f4, f4, f8, ud },
-    /* i4 */ { i4, i4, i4, i4, i8, f8, f8, f8, ud },
-    /* i8 */ { i8, i8, i8, i8, i8, f8, f8, f8, ud },
+    /* i4 */ { i4, i4, i4, i4, i8, f8, f4, f8, ud },
+    /* i8 */ { i8, i8, i8, i8, i8, f8, f4, f8, ud },
     /* f2 */ { f2, f2, f4, f8, f8, f2, f4, f8, ud },
-    /* f4 */ { f4, f4, f4, f8, f8, f4, f4, f8, ud },
+    /* f4 */ { f4, f4, f4, f4, f4, f4, f4, f8, ud },
```

Comment:
So float array + int/long array -> float array? This seems reasonable, but it's different from NumPy, right?

Reply:
That's right.